test/decl/ext/extensions.swift

```diff
@@ -104,24 +104,51 @@ var c = C()
 var x = c.p1
 c.p1 = 1
 
+// Reject extension of existential type
+
+protocol P3 {}
+
+extension any P3 {
+// expected-error@-1 {{extension of type 'any P3' is not supported}}
+// expected-note@-2 {{did you mean to extend 'P3' instead?}}
+}
+
 // Reject extension of nominal type via inferred associated type
-protocol P3 {
+protocol P4 {
   associatedtype Assoc
   func foo() -> Assoc
 }
 
-struct X3 : P3 {
+struct X4 : P4 {
 }
 
-extension X3.Assoc {
-// expected-error@-1 {{extension of type 'X3.Assoc' (aka 'Int') must be declared as an extension of 'Int'}}
+extension X4.Assoc {
+// expected-error@-1 {{extension of type 'X4.Assoc' (aka 'Int') must be declared as an extension of 'Int'}}
 // expected-note@-2 {{did you mean to extend 'Int' instead?}}
 }
 
-extension X3 {
+extension X4 {
   func foo() -> Int { return 0 }
 }
 
+// Reject extension of nominal type via typealias with dependent underlying type
+
+struct Nest<Egg> { typealias Contents = Egg }
+struct Tree { 
+  typealias LimbContent = Nest<Int> 
+  typealias BoughPayload = Nest<Nest<Int>>
+}
+
+extension Tree.LimbContent.Contents {
+  // expected-error@-1 {{extension of type 'Tree.LimbContent.Contents' (aka 'Int') must be declared as an extension of 'Int'}}
+  // expected-note@-2 {{did you mean to extend 'Int' instead?}} {{11-36=Int}}
+}
+
+extension Tree.BoughPayload.Contents {
+  // expected-error@-1 {{extension of type 'Tree.BoughPayload.Contents' (aka 'Nest<Int>') must be declared as an extension of 'Nest<Int>'}}
+  // expected-note@-2 {{did you mean to extend 'Nest<Int>' instead?}}  {{11-37=Nest<Int>}}
+}
+
```

Comment:
Any reason for moving this here from below?

Reply:
I moved it here just because it is related to the things above (tests just a slightly different instance of the same kind of diagnostic). Please let me know if it is preferred to minimize such unnecessary changes, or if the old location had a reason.

Reply:
Usually new tests go at the end of the file, so that's probably why it was at that specific location.

Reply:
Good to know. Done!